inst/training_202307/202307.qmd

```diff
@@ -429,16 +484,12 @@ None of the test result is _stable_:
 
     This is the most instable environment changed by pushes to any of the dependencies. Also: the main branch is usually not guaranteed to be correct and you might pull broken package version.
 
-# A few caveats:
-
-What it is not about:
-
-- This does not perform matrix testing of _all_ dependent package versions.
-- This only looks into package versions - in particular: it does not check R versions or system architectures.
+Therefore, this test should be triggered periodically and not on code change (unless changes includes dependencies). Triggering it on change might give a false message that a failure stems from code changes.
 
 # Future work:
 
```

Comment:
idea: concept of "min version to check", usage:

1. coalesce on a cohort of direct dependencies. 
2. look in DESC files for those dependencies to see versioned requirements they have  on *their* dependencies
3. set the min versions to even consider for each recursive dep to be the max of all versioned dependencies on it from the cohort from (1), if any.

Should dramatically reduce the number of invalid version checks done in the min_isolated strategy

Reply:
I agree. Recursive search should be probably the next step. This will get us closer to actual minimal version environment as well as remove redundant nodes in the dependency tree.

Reply:
Added this to the future work